tests/devices/test_default_clifford.py

```diff
@@ -94,38 +93,276 @@ def circuit_fn():
+@pytest.mark.parametrize("shots", [None, 8192])
+@pytest.mark.parametrize(
+    "ops",
+    [
+        None,
+        qml.PauliY(0),
+        qml.PauliX(0) @ qml.PauliY(1),
+        qml.PauliX(0) @ qml.PauliY(1) @ qml.PauliZ(2),
+    ],
+)
+def test_meas_probs(tableau, shots, ops):
+    """Test if probabilities are returned in the clifford device."""
+
+    dev_c = qml.device("default.clifford", tableau=tableau, shots=shots, seed=24)
+    dev_q = qml.device("default.qubit")
+
+    def circuit_fn():
+        for wire in range(3):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.probs(op=ops) if ops else qml.probs(wires=[0, 1])
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    gotten_probs, target_probs = qnode_clfrd(), qnode_qubit()
+    if gotten_probs.shape[-1] != target_probs.shape[-1]:
+        target_probs = qml.math.sum(target_probs.reshape((2,) * len(ops.wires)), axis=-1).reshape(
+            1, -1
+        )
+
+    assert qml.math.allclose(gotten_probs, target_probs, atol=1e-2 if shots else 1e-8)
+
+
+def test_meas_probs_large():
+    """Test if probabilities are returned in the clifford device with target basis states"""
+
+    dev_c = qml.device("default.clifford", seed=24)
+
+    def circuit_fn1():
+        for wire in range(30):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.probs(wires=range(1))
+
+    assert qml.QNode(circuit_fn1, dev_c)().shape == (2,)
+
+    def circuit_fn2(state):
+        for wire in range(30):
+            qml.PauliX(wire)
+            qml.PauliY(wire)
+            qml.PauliZ(wire)
+        return qml.expval(qml.Projector(state, wires=[0, 1]))
+
+    qnode_clfrd = qml.QNode(circuit_fn2, dev_c)
+
+    basis_states = np.array([[0, 0], [1, 0], [0, 1], [1, 1]])
+    probs1 = [qnode_clfrd(state) for state in basis_states]
+    probs2 = [1.0, 0.0, 0.0, 0.0]
+    assert qml.math.allclose(probs1, probs2)
+
+
+@pytest.mark.parametrize("shots", [1024, 4096])
+@pytest.mark.parametrize(
+    "ops",
+    [None, qml.PauliY(0), qml.PauliX(0) @ qml.PauliY(1)],
+)
+def test_meas_counts(shots, ops):
+    """Test if counts are returned with shots given in the clifford device."""
+
+    dev_c = qml.device("default.clifford", shots=shots, seed=24)
+    dev_q = qml.device("default.qubit", shots=shots, seed=24)
+
+    def circuit_fn():
+        qml.PauliX(0)
+        qml.PauliX(1)
+        return qml.counts(op=ops) if ops else qml.counts(wires=[0, 1])
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    counts_clfrd = qnode_clfrd()
+    counts_qubit = qnode_qubit()
+
+    assert list(counts_clfrd.keys()) == list(counts_qubit.keys())
+
+    for k1, k2 in zip(counts_clfrd, counts_clfrd):
+        assert qml.math.abs(counts_clfrd[k1] - counts_qubit[k2]) / shots < 5.0
```

Comment:
What is the 5 here?

Reply:
Oops, it was supposed to be % threshold, fixed it.